lib/sensors/gyroscope.cpp

```diff
@@ -0,0 +1,117 @@
+{
+  if (axis == core::GyroscopeAxis::kX) {
+    const auto gyroscope_x_high_byte = i2c_.readByte(kDefaultGyroscopeAddress, kDataXHigh);
+    if (!gyroscope_x_high_byte) {
+      logger_.log(
+        core::LogLevel::kFatal, "Failed to, read gyroscope X-axis high at channel %d", channel_);
+      return std::nullopt;
+    }
+    const auto gyroscope_x_low_byte = i2c_.readByte(kDefaultGyroscopeAddress, kDataXLow);
+    if (!gyroscope_x_low_byte) {
+      logger_.log(
+        core::LogLevel::kFatal, "Failed to, read gyroscope X-axis low at channel %d", channel_);
+      return std::nullopt;
+    }
+    const auto x_axis = ((gyroscope_x_high_byte.value() << 8) | gyroscope_x_low_byte.value());
+    logger_.log(
+      core::LogLevel::kDebug, "Successfully read x-axis from gyroscope at channel %d", channel_);
+    return x_axis;
+  } else if (axis == core::GyroscopeAxis::kY) {
+    const auto gyroscope_y_high_byte = i2c_.readByte(kDefaultGyroscopeAddress, kDataYHigh);
+    if (!gyroscope_y_high_byte) {
+      logger_.log(
+        core::LogLevel::kFatal, "Failed to, read gyroscope Y-axis high at channel %d", channel_);
+      return std::nullopt;
+    }
+    const auto gyroscope_y_low_byte = i2c_.readByte(kDefaultGyroscopeAddress, kDataYLow);
+    if (!gyroscope_y_low_byte) {
+      logger_.log(
+        core::LogLevel::kFatal, "Failed to, read gyroscope Y-axis low at channel %d", channel_);
+      return std::nullopt;
+    }
+    const auto y_axis = ((gyroscope_y_high_byte.value() << 8) | gyroscope_y_low_byte.value());
+    logger_.log(
+      core::LogLevel::kDebug, "Successfully read y-axis from gyroscope at channel %d", channel_);
+    return y_axis;
+  } else if (axis == core::GyroscopeAxis::kZ) {
+    const auto gyroscope_z_high_byte = i2c_.readByte(kDefaultGyroscopeAddress, kDataZHigh);
+    if (!gyroscope_z_high_byte) {
+      logger_.log(
+        core::LogLevel::kFatal, "Failed to, read gyroscope Z-axis high at channel %d", channel_);
+      return std::nullopt;
+    }
+    const auto gyroscope_z_low_byte = i2c_.readByte(kDefaultGyroscopeAddress, kDataZLow);
+    if (!gyroscope_z_low_byte) {
+      logger_.log(
+        core::LogLevel::kFatal, "Failed to, read gyroscope Z-axis low at channel %d", channel_);
+      return std::nullopt;
+    }
+    const auto z_axis = ((gyroscope_z_high_byte.value() << 8) | gyroscope_z_low_byte.value());
+    logger_.log(
+      core::LogLevel::kDebug, "Successfully read z-axis from gyroscope at channel %d", channel_);
+    return z_axis;
+  } else {
+    logger_.log(core::LogLevel::kFatal,
+                "Failed to, read gyroscope at channel %d, axis parameters are invalid types",
+                channel_);
+    return std::nullopt;
+  }
+}
+
+std::optional<Gyroscope> Gyroscope::create(core::ILogger &logger,
+                                           io::II2c &i2c,
+                                           const std::uint8_t channel)
+{
+  const core::Result write_result1
```

Comment:
be more descriptive with the name + add a comment to describe what this is even doing (i.e. what setting does it enforce?)